File: go/environment.go
package applereceipt

// Environment is an App Store environment, spelled exactly as Apple's
// JWS claims and receipt attributes spell it.
//
// Environment routing is an accept-set, not a single value (PLAN.md D3):
// App Review runs production builds against sandbox, so a verifier that
// hard-failed on one environment would reject purchases during review.
type Environment string

// The four environments Apple uses.
const (
	EnvironmentProduction   Environment = "Production"
	EnvironmentSandbox      Environment = "Sandbox"
	EnvironmentXcode        Environment = "Xcode"
	EnvironmentLocalTesting Environment = "LocalTesting"
)

func (e Environment) known() bool {
	switch e {
	case EnvironmentProduction, EnvironmentSandbox, EnvironmentXcode, EnvironmentLocalTesting:
		return true
	}
	return false
}

// String returns the claim spelling.
func (e Environment) String() string { return string(e) }

File: php/src/Environment.php
<?php

declare(strict_types=1);

namespace EminDeniz99\ApplePurchaseReceiptVerifier;

/**
 * The App Store environment a payload was issued in, spelled exactly as the
 * JWS claim spells it (`environment` on a transaction, `receiptType` on an
 * AppTransaction).
 *
 * Verification takes an accept-*set* rather than a single value: App Review
 * runs production builds against sandbox, so an endpoint that hard-failed on
 * anything but Production would reject purchases during review (PLAN.md D3).
 */
enum Environment: string
{
    case Production = 'Production';
    case Sandbox = 'Sandbox';
    case Xcode = 'Xcode';
    case LocalTesting = 'LocalTesting';
}
